prime-router/src/main/kotlin/ActionMessages.kt

```diff
@@ -140,4 +140,26 @@ class InvalidReportMessage(override val message: String) : GenericActionLogDetai
  * A [message] for invalid translation operation.
  */
 class InvalidTranslationMessage(override val message: String) :
-    GenericActionLogDetail(message, ActionLogScope.translation)
\ No newline at end of file
+    GenericActionLogDetail(message, ActionLogScope.translation)
+
+/**
+ * A [message] for entire duplicate report submission
+ */
+class DuplicateFileMessage(val payloadName: String?, val allItemsDupe: Boolean) : ActionLogDetail {
+    override val scope = ActionLogScope.report
+    override val message: String get() {
+        var msg = if (allItemsDupe) "All items in this file are duplicates." else "Duplicate file detected."
```

Comment:
1. Maybe change file to report which is the term we use.
2. I am confused by these messages.  It sounds like the first one is for when a complete report is duplicate.  But the second sounds like an item or more, but the message says file.  Can you clarify?

Reply:
Updated this, as we are no longer doing overall submission-level dupe checking anyway.